packages/teraslice/src/lib/cluster/services/cluster/backends/kubernetesV2/k8s.ts

```
@@ -536,19 +518,28 @@ export class K8s {
      * of workers.
      * @param  {String} exId       exId of execution to scale
      * @param  {number} numWorkers number of workers to scale by
-     * @param  {String} op         Scale operation: `set`, `add`, `remove`
+     * @param  {ScaleOp} op         Scale operation: `set`, `add`, `remove`
      * @return {Object}            Body of patch response.
      */
-    async scaleExecution(exId: string, numWorkers: number, op: string) {
-        let newScale;
+    async scaleExecution(exId: string, numWorkers: number, op: ScaleOp): Promise<k8s.V1Deployment> {
+        let newScale: number;
+        const selector = `app.kubernetes.io/component=worker,teraslice.terascope.io/exId=${exId}`;
 
         this.logger.info(`Scaling exId: ${exId}, op: ${op}, numWorkers: ${numWorkers}`);
-        const listResponse = await this.list(`app.kubernetes.io/component=worker,teraslice.terascope.io/exId=${exId}`, 'deployments') as k8s.V1DeploymentList;
+        const listResponse = await this.list(selector, 'deployment');
         this.logger.debug(`k8s worker query listResponse: ${JSON.stringify(listResponse)}`);
 
         // the selector provided to list above should always result in a single
         // deployment in the response.
-        // TODO: test for more than 1 and error
+        if (listResponse.items.length === 0) {
+            const msg = `Teraslice deployment matching the following selector was not found: ${selector} (retriable)`;
+            this.logger.warn(msg);
+            throw new TSError(msg, { retryable: true });
+        } else if (listResponse.items.length > 1) {
+            throw new TSError(`Unexpected number of Teraslice deployments matching the following selector: ${selector}`, {
```

So what is the consequence of this being thrown?

---

This function can only be reached from a post to `/jobs/:jobId/_workers` or `/ex/:exId/_workers`. There are no catches along that route, so it should return the error back to the user.